### ISSUE TITLE
Install failure: Does installing buck require GLIBC_2_14?

### ISSUE THREAD

**Issue by jhdub23** [facebook/buck#1813]
Is glibc_2_14 a requirement for buck, or is there some issue with ANT or other depencency?  I get the error below when trying to install buck on a CentOS6 system:

% ./bin/buck build buck
2018-03-21 10:59:35,928 [WARNING][buck_tool.py:412] Not using buckd because daemon failed to start.
Exception in thread "main" java.lang.UnsatisfiedLinkError: /data3/jhwest/pkgs/src/buck/buck-out/tmp/buck_run.QL_R54/jna--1160585235/jna601736662054321942.tmp: /lib64/libc.so.6: version `GLIBC_2.14' not found (required by /data3/jhwest/pkgs/src/buck/buck-out/tmp/buck_run.QL_R54/jna--1160585235/jna601736662054321942.tmp)
	at java.lang.ClassLoader$NativeLibrary.load(Native Method)
	at java.lang.ClassLoader.loadLibrary0(ClassLoader.java:1941)
	at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1824)
	at java.lang.Runtime.load0(Runtime.java:809)
	at java.lang.System.load(System.java:1086)
	at com.sun.jna.Native.loadNativeDispatchLibraryFromClasspath(Native.java:947)
	at com.sun.jna.Native.loadNativeDispatchLibrary(Native.java:922)
	at com.sun.jna.Native.<clinit>(Native.java:190)
	at com.sun.jna.Pointer.<clinit>(Pointer.java:54)
	at com.facebook.buck.cli.Main.<clinit>(Main.java:229)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.facebook.buck.cli.bootstrapper.ClassLoaderBootstrapper.main(ClassLoaderBootstrapper.java:55)

**Comment by styurin** [comment]
Can you run other java applications? Looks like something's wrong with JDK/JRE.

**Comment by jhdub23** [comment]
I can build and run a simple HelloWorld application.

"ant -diagnostics" gives:

java.runtime.name : OpenJDK Runtime Environment
sun.boot.library.path : /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.161-3.b14.el6_9.x86_64/jre/lib/amd64
java.vm.version : 25.161-b14

**Comment by styurin** [comment]
Have you tried to google "/lib64/libc.so.6: version `GLIBC_2.14' not found"? :)

**Comment by jhdub23** [comment]
First thing I tried.  Scoured the web.  Closest I came were for other applications:

https://github.com/tensorflow/tensorflow/issues/14208
https://github.com/java-native-access/jna/issues/853

This led me to believe that buck has java code that is directly using GLIB_2_14 features, via the JNI.  I'm from the C/C++ world and don't use JAVA, so I'm not familiar with how JAVA interacts with the system.

**Comment by styurin** [comment]
We actually have a dependency on JNA library (https://github.com/java-native-access/jna), looks like it requires glibc 2.14.

**Comment by jhdub23** [comment]
Perhaps the requirement of glibc 2.14 should be in the installation notes.  I finally got buck to work on our CentOS6 system.  I used linuxbrew to install the latest tool chain with an updated glibc.  Making things work without using LD_LIBRARY_PATH (which makes all the existing tools break) was a bit tricky.  I had to use "brew edit ant", "brew edit buck" to add additional dependencies (glibc, python2, jdk@8).
